packages/block-library/src/columns/column.js

```diff
@@ -1,12 +1,66 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
 /**
  * WordPress dependencies
  */
 import { Path, SVG } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { InnerBlocks } from '@wordpress/block-editor';
+import { InnerBlocks, BlockControls, BlockVerticalAlignmentToolbar } from '@wordpress/block-editor';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { compose } from '@wordpress/compose';
 
 export const name = 'core/column';
 
+const ColumnEdit = ( { attributes, updateAlignment } ) => {
+	const { verticalAlignment } = attributes;
+
+	const classes = classnames( 'block-core-columns', {
+		[ `is-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
+	} );
+
+	const onChange = ( alignment ) => updateAlignment( alignment );
+
+	return (
+		<div className={ classes }>
+			<BlockControls>
+				<BlockVerticalAlignmentToolbar
+					onChange={ onChange }
+					value={ verticalAlignment }
+				/>
+			</BlockControls>
+			<InnerBlocks templateLock={ false } />
+		</div>
+	);
+};
+
+const edit = compose(
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockRootClientId } = select( 'core/editor' );
+
+		return {
+			parentColumsBlockClientId: getBlockRootClientId( clientId ),
+		};
+	} ),
+	withDispatch( ( dispatch, { clientId, parentColumsBlockClientId } ) => {
+		return {
+			updateAlignment( alignment ) {
+				// Update self...
+				dispatch( 'core/editor' ).updateBlockAttributes( clientId, {
+					verticalAlignment: alignment,
+				} );
+
+				// Reset Parent Columns Block
+				dispatch( 'core/editor' ).updateBlockAttributes( parentColumsBlockClientId, {
```

Comment:
Is it necessary to update the block if it's not guaranteed that anything will change (i.e. what about if the parent block had no vertical alignment)?